File: src/002/p002.swift
func solve() -> Int {
    var a = 1
    var b = 2
    var sum = 0

    while b < 4000000 {
        if b % 2 == 0 {
            sum += b
        }
        let temp = b
        b = a + b
        a = temp
    }

    return sum
}

let result = solve()
print(result)

File: src/002/p002.pl
#!/usr/bin/env perl
use strict;

sub solve {
    my $a = 1;
    my $b = 2;
    my $sum = 0;
    while ($b < 4000000) {
        $sum += $b if $b % 2 == 0;
        my $temp = $b;
        $b += $a;
        $a = $temp;
    }
    return $sum;
}

my $result = solve;
print($result);
